### PR TITLE
#153 improve random data set implementation

### PR DESCRIPTION
**Pull request by oomelianchuk**

### CODE REVIEW

File: src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java [hunk 1]
```diff
@@ -191,20 +191,18 @@ else if (!classDataSetAnnotations.isEmpty())
             dataSetAnnotations = classDataSetAnnotations;
         }
 
+        // we've gathered some instructions to override
```
Comment (occupant23) [inline src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java:194]: please make this comment more clear or put it to the correct place

File: src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java [hunk 2]
```diff
@@ -266,6 +250,21 @@ else if (randomSetAmount > 0)
                 }
             }
         }
+        RandomDataSets methodRandomDataSetsAnnotation = method.getAnnotation(RandomDataSets.class);
```
Comment (occupant23) [inline src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java:253]: please ad somme comments for the random data set logic to help out our future me

File: src/test/java/com/xceptance/neodymium/tests/TestDataStatementTest.java [hunk 3]
```diff
@@ -401,6 +404,27 @@ public void testMixRandomAndValueDataSets()
         checkPass(result, 2, 0);
     }
```
Comment (occupant23) [inline src/test/java/com/xceptance/neodymium/tests/TestDataStatementTest.java:406]: please add a test case for overriding the class level using the method level.
this should have at least two method and one of the is using the overriding mechanic while the other is using the class level

**Review by occupant23 — CHANGES_REQUESTED** [review]

**Review by occupant23 — CHANGES_REQUESTED** [review]
please see the comments

File: src/test/java/com/xceptance/neodymium/tests/TestDataStatementTest.java (hunk 3 above)
Comment (oomelianchuk) [reply src/test/java/com/xceptance/neodymium/tests/TestDataStatementTest.java:407]: Done

File: src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java (hunk 2 above)
Comment (oomelianchuk) [reply src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java:254]: Done

File: src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java (hunk 1 above)
Comment (oomelianchuk) [reply src/main/java/com/xceptance/neodymium/module/statement/testdata/TestdataStatement.java]: Comment removed

File: src/main/java/com/xceptance/neodymium/module/statement/testdata/RandomDataSets.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation can be used to limit and override data set execution for an entire class or at the same time for a
```
[the remark this answers is not in the document]
Comment (oomelianchuk) [reply src/main/java/com/xceptance/neodymium/module/statement/testdata/RandomDataSets.java]: Done

**Review by occupant23 — APPROVED** [review]
please see the comments

**Pull request merged**